carmirror/car.go

```diff
@@ -10,10 +10,12 @@ import (
 	"github.com/ipld/go-car"
 )
 
-func AddAllFromCarReader(ctx context.Context, bapi coreiface.BlockAPI, r io.Reader, progCh chan cid.Cid) (int, error) {
+func AddAllFromCarReader(ctx context.Context, bapi coreiface.BlockAPI, r io.Reader, progCh chan cid.Cid) (int, []cid.Cid, error) {
+	log.Debugf("AddAllFromCarReader")
+	cids := []cid.Cid{}
```

Comment:
Could we allocate a capacity of at least len(rdr.Header.Roots) entries in the cid slice in order to speed things up?

Reply:
Fabulous. Thank you!